cfdm/mixin/fielddomain.py

```diff
@@ -0,0 +1,1219 @@
+            if data is None:
+                continue
+
+            x = getattr(data, f"get_{component}")(None)
+            if x is None:
+                continue
+
+            out.append(x)
+
+        return out
+
+    def _get_coordinate_geometry_variables(self, component):
+        """Return the list of variables for the geometry coordinates.
+
+        :Parameters:
+
+            component: `str`
+
+        :Returns:
+
+            `list'
+
+        """
+        out = []
+        for construct in self.coordinates(todict=True).values():
+            x = getattr(construct, f"get_{component}")(None)
+            if x is None:
+                continue
+
+            out.append(x)
+
+        return out
+
+    def _filter_interface(
+        self,
+        _ctypes,
+        _method,
+        identities,
+        cached=None,
+        construct=False,
+        key=False,
+        item=False,
+        default=None,
+        _identity_config={},
+        **filter_kwargs,
+    ):
+        """An optimised interface to `Constructs.filter`.
+
+        .. versionadded:: (cfdm) 1.8.9.0
+
+        :Parameters:
+
+            _ctypes: `tuple` of `str`
+                The construct types to restrict the selection to.
+
+            _method: `str`
+                The name of the calling method.
+
+            identities: `tuple`
+                Select constructs that have an identity, defined by
+                their `!identities` methods, that matches any of the
+                given tuple values.
+
+                {{value match}}
+
+            {{todict: `bool`, optional}}
+
+            {{cached: optional}}
+
+            {{filter_kwargs: optional}}
+
+        :Returns:
+
+            `Constructs` or `dict` or *cached*
+                If *cached* is `None`, return the selected
+                constructs. If *cached* is not `None`, return
+                *cached*.
+
+        """
+        if cached is not None:
+            return cached
+
+        if not _ctypes:
+            kwargs = filter_kwargs
+        else:
+            # Ensure that filter_by_types is the first filter
+            # applied, as it's the cheapest
+            if not (identities or filter_kwargs):
+                # This a very common pattern for which calling
+                # filter_by_type directly is faster
+                if construct:
+                    todict = True
+                else:
+                    todict = False
+
+                c = self.constructs.filter_by_type(*_ctypes, todict=todict)
+
+                if not construct:
+                    # Return Constructs or dict
+                    return c
+
+                # Return construct, or key, or both, or default
+                n = len(c)
+                if n == 1:
+                    k, construct = c.popitem()
+                    if key:
+                        return k
+
+                    if item:
+                        return k, construct
+
+                    return construct
```

Comment:
Yes, they should not both be True - I'll add a note to the effect that if `key=True` then item is ignored. I think that `key=True` is more often used, so that's why I put it first.

Reply:
Great, thanks for the clarification and the new note in the docstring which makes the context very clear.